packages/animation-curve-block/src/components/AnimationCurveFlyout.tsx

```diff
@@ -0,0 +1,228 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useEffect, useState } from 'react';
+
+import {
+    ButtonEmphasis,
+    ButtonStyle,
+    Dropdown,
+    Flyout,
+    FlyoutFooter,
+    FlyoutPlacement,
+    IconCaretRight16,
+    IconCheckMark16,
+    TextInput,
+    TextInputType,
+    Validation,
+} from '@frontify/fondue';
+
+import {
+    AnimationCurve,
+    AnimationCurveFlyoutProps,
+    AnimationCurveParametersPatch,
+    AnimationCurveType,
+    AnimationFunctionPatch,
+    defaultAnimationCurveTypeValues,
+} from '../types';
+import { DROPDOWN_MENU_ITEMS } from '../constants';
+import { AnimationCanvas } from './';
+
+export const AnimationCurveFlyout = ({
+    animationCurve,
+    isFlyoutOpen,
+    lineColor,
+    endpointColor,
+    gridColor,
+    onSave,
+    onCancel,
+    onOpenChange,
+    onAnimationCurveChange,
+}: AnimationCurveFlyoutProps) => {
+    const [x1Validation, setX1Validation] = useState<Validation>(Validation.Default);
+    const [x2Validation, setX2Validation] = useState<Validation>(Validation.Default);
+    const [localAnimationFunction, setLocalAnimationFunction] = useState(animationCurve.animationFunction);
+
+    useEffect(() => {
+        if (localAnimationFunction.parameters.x1) {
+            if (localAnimationFunction.parameters.x1 < 0 || localAnimationFunction.parameters.x1 > 1) {
+                setX1Validation(Validation.Error);
+            } else {
+                setX1Validation(Validation.Default);
+            }
+        }
+        if (localAnimationFunction.parameters.x2) {
+            if (localAnimationFunction.parameters.x2 < 0 || localAnimationFunction.parameters.x2 > 1) {
+                setX2Validation(Validation.Error);
+            } else {
+                setX2Validation(Validation.Default);
+            }
+        }
+    }, [localAnimationFunction]);
+
+    const handleSave = () => {
+        onSave(animationCurve.id, { ...animationCurve });
```

Comment:
If previous animation curve has title, the newly created one will copy it. I think this is a bug.